app/models/concerns/translatable.rb

```diff
@@ -47,23 +47,69 @@ def attr_translatable(*attributes)
     end
   end
 
-  def get_translation(attribute)
-    translation = translations.find_by(translatable_attribute: attribute, locale: I18n.locale)
-    translation = get_translations(attribute).first  unless translation
-    return associated_translation(translation) if translation
-    nil
+  def get_translation(attribute, locale: I18n.locale, exact_match: false)
+    # important to filter against the association held in memory instead of performing a db lookup
+    # in order to redisplay potentially transient values
+    result = translations.find do |t|
+      t.translatable_attribute.to_sym == attribute.to_sym && t.locale.to_sym == locale.to_sym
+    end
+    unless result || exact_match
+      # fall back to first translations of any locale if desired locale not present
+      # (except when fetching in order to perform update)
+      result = translations.find do |t|
+        t.translatable_attribute.to_sym == attribute.to_sym
+      end
+    end
+    result
   end
 
   def get_translations(attribute)
     translations.where(translatable_attribute: attribute)
   end
 
-  def set_translation(attribute, text, locale: I18n.locale)
-    translation = get_translation(attribute)
+  def used_locales
+    result = []
+    if respond_to?(:division)
```

Comment:
I tend to prefer duck typing to using `respond_to?` per Sandy Metz's advice. But probably fine for now. Just something to keep in mind.